### ISSUE TITLE
Respect project `root` option

### ISSUE THREAD

**Issue by atinux** [brattonross/vite-plugin-voie#10]
First, nice work on this module 👏 

Actually when using the [root](https://github.com/vitejs/vite/blob/master/src/node/config.ts#L52) option in the `vite.config.ts`. I also have to define it in the `pagesDir` option.

I expect to be joined directly.

```ts
import { join } from 'path'
import voie from 'vite-plugin-voie'

export default {
  root: join(__dirname, 'app'),
   plugins: [
    voie({
      pagesDir: 'app/pages' // should be only pages
    })
  ]
}
```

**Comment by atinux** [comment]
I know @antfu fixed it for `vite-plugin-component`: https://github.com/antfu/vite-plugin-components/commit/60ed61054212047b8602819f8a93ef886784e235

**Comment by brattonross** [comment]
Thanks for your kind words. I decided to make this module because I missed the routing features of Nuxt, so thanks should also go to you and the Nuxt team for the inspiration 😅

This is a great suggestion, I'll take a look at it when I get the chance

**Comment by brattonross** [comment]
I have developed an initial working implementation of this, but it's a bit rough around the edges currently.

The fix that was implemented for `vite-plugin-component` as far as I can tell requires you to duplicate the vite `root` config in the component options, so I have gone for the same approach here. I can look at automatically resolving the root from the vite config once [this issue](https://github.com/vitejs/vite/issues/738) with vite is resolved.

Your config could now look something like:
```js
import { join } from 'path';
import voie from 'vite-plugin-voie';

const root = join(__dirname, 'app');

export default {
  root,
  plugins: [voie({ root, pagesDir: 'pages' })],
};
```
with your directory structure being something like:
```
/app
  /pages
    index.vue
  App.vue
  index.html
  main.js
package.json
vite.config.js
```
